components/ExpenseForm/index.js

```diff
@@ -93,78 +23,108 @@ function ExpenseForm() {
     return <h1> Error:{error.message} </h1>;
   }
 
-  const handleSubmit = async (event) => {
-    event.preventDefault();
-
-    // Handle form submission logic here
-    const formdata = new FormData(event.target);
-    const expData = Object.fromEntries(formdata);
-
-    const response = await fetch("/api/expenses", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(expData),
-    });
-
-    if (!response.ok) {
-      console.error(response.status);
-      return;
-    }
-
-    mutate();
-    event.target.reset();
-
-    Router.push("/");
-  };
-
   return (
     <FormContainer>
-      <FormTitle>Add Expense</FormTitle>
-      <form onSubmit={handleSubmit}>
-        <FormGroup>
-          <StyledInput
-            id="name"
-            name="name"
-            type="text"
-            placeholder="Title"
-            maxLength={100}
-            required
-          />
-        </FormGroup>
-        <FormGroup>
-          <StyledSelect id="categoryId" name="categoryId" required>
-            <option value="0"> Select Category</option>
-            {data.map((category, index) => (
-              <option key={index} value={category._id}>
-                {category.name}
-              </option>
-            ))}
-          </StyledSelect>
-        </FormGroup>
-        <FormGroup>
-          <StyledTextarea
-            id="description"
-            name="description"
-            placeholder="Description"
-            rows="4"
-            maxLength={500}
-          />
-        </FormGroup>
-        <FormGroup>
-          <StyledInput
-            type="number"
-            id="amount"
-            name="amount"
-            placeholder="Amount"
-            min={0}
-            required
-          />
-        </FormGroup>
+      <FormTitle>{isEditMode ? "Edit Expense" : "Add Expense"}</FormTitle>
+      <form onSubmit={onSubmit}>
+        {isEditMode ? (
+          <>
+            <FormGroup>
+              <StyledInput
+                id="name"
+                name="name"
+                type="text"
+                placeholder="Title"
+                maxLength={100}
+                required
+                defaultValue={expense.name}
+              />
+            </FormGroup>
+            <FormGroup>
+              <StyledSelect
+                id="categoryId"
+                name="categoryId"
+                required
+                defaultValue={expense.categoryId[0]._id}
+              >
+                <option value="0"> Select Category</option>
+                {data.map((data, index) => (
+                  <option key={index} value={data._id}>
```

Comment:
if you have access to the id as a unique identifier, better to use this for the key

```suggestion
                  <option key={data._id} value={data._id}>
```